### ISSUE TITLE
New rule: Function paramater should not be nullable

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#4377]
## Expected Behavior of the rule
Some times I found functions that allows a null parameter to then assert that it's not null. Those functions shouldn't accept a null parameter, they should force the caller to take the decission to force the non-null or don't cal it.

Non-compilant
```kotlin
fun foo(a: String?) {
  println(a!!)
}
```

This function should be declared like this:
```kotlin
fun foo(a: String) {
  println(a)
}
```

## Context
This kind of errors are added with the java to kotlin translator. And some times after a few refactor you can end up with code like this.

**Comment by severn-everett** [comment]
I'll add this in when PR #4379 is integrated, as it looks like it falls within the same purview of that rule.

**Comment by cortinico** [comment]
Closed by #4431
